Client/Frontend/Login Management/LoginListViewModel.swift

```diff
@@ -0,0 +1,192 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Storage
+import Shared
+
+// MARK: - Main View Model
+/// View Model to handle logic for LoginListViewController.
+final class LoginListViewModel {
+    
+    let profile: Profile
+    fileprivate var activeLoginQuery: Deferred<Maybe<[LoginRecord]>>? = nil
+    var dataSource: LoginListDataSourceViewModel
+    var isDuringSearchControllerDismiss = false
+
+    init(profile: Profile, searchController: UISearchController) {
+        self.profile = profile
+        self.dataSource = LoginListDataSourceViewModel(searchController)
+    }
+
+    func loadLogins(_ query: String? = nil, loginDataSource: LoginDataSource) {
+        // Fill in an in-flight query and re-query
+        activeLoginQuery?.fillIfUnfilled(Maybe(success: []))
+        activeLoginQuery = queryLogins(query ?? "")
+        activeLoginQuery! >>== dataSource.setLogins
+    }
+    
+    /// Searches SQLite database for logins that match query.
+    /// Wraps the SQLiteLogins method to allow us to cancel it from our end.
+    func queryLogins(_ query: String) -> Deferred<Maybe<[LoginRecord]>> {
+        let deferred = Deferred<Maybe<[LoginRecord]>>()
+        profile.logins.searchLoginsWithQuery(query) >>== { logins in
+            deferred.fillIfUnfilled(Maybe(success: logins.asArray()))
+            succeed()
+        }
+        return deferred
+    }
+
+    // MARK: - UX Constants
+    struct LoginListUX {
+        static let RowHeight: CGFloat = 58
+        static let SearchHeight: CGFloat = 58
+        static let selectionButtonFont = UIFont.systemFont(ofSize: 16)
+        static let NoResultsFont = UIFont.systemFont(ofSize: 16)
+        static let NoResultsTextColor = UIColor.Photon.Grey40
+    }
+
+    // MARK: - Sub View Models
+    final class LoginListDataSourceViewModel {
```

Comment:
Could we move it out of the actual view model and not call it a ViewModel as there should be just 1 VM.

Reply:
If you would rather like to keep it in the viewmodel thats also fine but then you'll need to move items out of the final class directly inside viewmodel. 

Go by the rule of having "Single View Model"